presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/rtree/TestFlatbush.java

```diff
@@ -0,0 +1,300 @@
+
+    //  2 intersecting polygons: A and B
+    private static final OGCGeometry POLYGON_A = OGCGeometry.fromText("POLYGON ((0 0, -0.5 2.5, 0 5, 2.5 5.5, 5 5, 5.5 2.5, 5 0, 2.5 -0.5, 0 0))");
+    private static final OGCGeometry POLYGON_B = OGCGeometry.fromText("POLYGON ((4 4, 3.5 7, 4 10, 7 10.5, 10 10, 10.5 7, 10 4, 7 3.5, 4 4))");
+
+    // A set of points: X in A, Y in A and B, Z in B, W outside of A and B
+    private static final OGCGeometry POINT_X = new OGCPoint(new Point(1.0, 1.0), null);
+    private static final OGCGeometry POINT_Y = new OGCPoint(new Point(4.5, 4.5), null);
+    private static final OGCGeometry POINT_Z = new OGCPoint(new Point(6.0, 6.0), null);
+    private static final OGCGeometry POINT_W = new OGCPoint(new Point(20.0, 20.0), null);
+
+    private static <T extends HasExtent> List<T> findIntersections(Flatbush<T> rtree, Rectangle rectangle)
+    {
+        List<T> results = new ArrayList<>();
+        rtree.findIntersections(rectangle, results::add);
+        return results;
+    }
+
+    /*
+     * Asserts the two lists of Rectangles are equal after sorting.
+     */
+    private static <T> void assertEqualsSorted(List<T> actual, List<T> expected, Comparator<T> comparator)
+    {
+        List<T> actualSorted = actual.stream().sorted(comparator).collect(toImmutableList());
+        List<T> expectedSorted = expected.stream().sorted(comparator).collect(toImmutableList());
+
+        assertEquals(actualSorted, expectedSorted);
+    }
+
+    @Test
+    public void testEmptyFlatbush()
+    {
+        Flatbush<Rectangle> rtree = new Flatbush<>(new Rectangle[] {});
+        assertEquals(findIntersections(rtree, EVERYTHING), ImmutableList.of());
+    }
+
+    @Test
+    public void testSingletonFlatbush()
+    {
+        List<Rectangle> items = new ArrayList<>();
+        items.add(new Rectangle(0, 0, 1, 1));
+        Flatbush<Rectangle> rtree = new Flatbush<>(items.toArray(new Rectangle[] {}));
+
+        assertEquals(findIntersections(rtree, EVERYTHING), items);
+        // hit
+        assertEquals(findIntersections(rtree, new Rectangle(1, 1, 2, 2)), items);
+        // miss
+        assertEquals(findIntersections(rtree, new Rectangle(-1, -1, -0.1, -0.1)), ImmutableList.of());
+    }
+
+    @Test
+    public void testSingletonFlatbushXY()
+    {
+        // Because mixing up x and y is easy to do...
+        List<Rectangle> items = new ArrayList<>();
+        items.add(new Rectangle(0, 10, 1, 11));
+        Flatbush<Rectangle> rtree = new Flatbush<>(items.toArray(new Rectangle[] {}));
+
+        // hit
+        assertEquals(findIntersections(rtree, new Rectangle(1, 11, 2, 12)), items);
+        // miss
+        assertEquals(findIntersections(rtree, new Rectangle(11, 1, 12, 2)), ImmutableList.of());
+    }
+
+    @Test
+    public void testDoubletonFlatbush()
+    {
+        // This is the smallest Rtree with height > 1
+        // Also test for some degeneracies
+        List<Rectangle> items = new ArrayList<>();
```

Comment:
use ImmutableList.of() and assertEqualsNoOrder

```
        Rectangle rect0 = new Rectangle(1, 1, 1, 1);
        Rectangle rect1 = new Rectangle(-1, -2, -1, -1);
        List<Rectangle> items = ImmutableList.of(rect0, rect1);

        Flatbush<Rectangle> rtree = new Flatbush<>(items.toArray(new Rectangle[] {}));

        List<Rectangle> allResults = findIntersections(rtree, EVERYTHING);
        assertEqualsNoOrder(allResults.toArray(), items.toArray());
```

ditto testTwoLevelFlatbush